### DISCUSSION TITLE
Put the Heli tab back in the default firmware

### DISCUSSION THREAD

**Discussion by NameOfTheDragon** [EdgeTX/edgetx#2913] [General]
Dear devs, what were you thinking? You removed the heli tab from the default firmware build, I presume as a convenience gain for those who don't fly helicopters. However, in the process, you have crippled the product and alienated a whole section of the user base for the sake of what seems like a very minimal gain.

I don't fly helis but as a professional developer, this seems to me to be a be a poor decision. End users should never be forced to compile firmware in order to get access to standard features, and certainly not when those features were always available in the past.

Perhaps I've interpreted this incorrectly but from the comments I am seeing in the community, I suspect not. As someone who doesn't fly helis I feel like it is my responsibility to speak up for them.

May I suggest that you put this feature back in for the time being at least, and maybe figure out a way to turn it off as a setting in the radio instead, for a future release?

**Comment by lshems** [comment]
The Heli page was like a wrench to start the engine in an electrical car. It was set up in a way that didn't fit in the general concept of opentx for several reasons, but not the good ones.
If you read the entire history of discussions you can see that several alternatives exist, including a custom lua mixer doing EXACTLY the same in a way that IS withing the concept of now edgetx.
Also, it is a very small minority of the Heli flyers that need such a mixer functionality as most fly with flight controllers doing the mixing.

I am no developer, fly no heli's, and wouldn't care less if it would be reintroduced as an option. Ik der the condition that it doesn't block resources on BW radio's (which it currently does in opentx).

But it would remain something completely out of place with better options to het it solved than a Heli page.

Just my two cents.

**Comment by gagarinlg** [comment]
The Heli tab will be a config option in the next release

**Comment by pfeerick** [comment]
And, for those who can't build for themselves, in addition to the multiple builds of the firmwares for all the various languages, there are heli and no-heli builds here => https://github.com/pfeerick/lang-firmwares/releases, which is linked in the release notes. This is in addition to the offer which was made available for every release since 2.5 for custom builds to be made, with preferred options.

**Comment by raphaelcoeffic** [comment]
@lshems i like the idea with the custom LUA mixer script. However, these are executed only every 100ms, AFAIK, which is a bit slow. I would to have something along these lines that could be executed much more often though.

**Reply by lshems** [reply]
The lua mixer script was made.

Normal mixes are evaluated every 30 ms, no?

Could you speed up custom mixers?

**Reply by raphaelcoeffic** [reply]
Normal mixer lines are evaluated as often as the sync period, which could be as fast as every 1-2ms.

**Reply by lshems** [reply]
Well, then they are no realistic option for a lot of usecases. 

When I see Heli aerobatics where they wank the Heli from left to right and on top and below within seconds, 100ms might be way off the needed latency.

So no go for the current lua mixers then for the Heli mixers.

Unless... And that is what I proposed before, allow 'prebuilt' custom mixers that are hardcoded in c and only use the mixer interface to set them up. 

This would put the Helipage exactly where it belongs: on a custom mixer with the correct inputs to adjust.

Just my two cents.

P.s. This would open up a lot of other useful stuff like flaperon mixers, vtail mixers, .... First create and test in lua. If needed for super fast mixing, create C++ fixed mixer setup.

Solves in principle every exotic use case. Think of yet throttle zero etc. Instead of messing around with gvars everywhere, for which there are still only 9.
